### PR TITLE
[DSM] [Playwright] PEPPER-1107 Medical Records Request Workflow test

### PR DESCRIPTION
**Pull request by aweng98**
[PEPPER-1107](https://broadworkbench.atlassian.net/browse/PEPPER-1107)

Adding one new test `medical-record-request.spec.ts`. Ran on Dev and Test env to ensure it works consistently.

Screenshot 2023-11-16 at 3 28 04 PM

[PEPPER-1107]: https://broadworkbench.atlassian.net/browse/PEPPER-1107?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: playwright-e2e/dsm/component/date-picker.ts [hunk 1]
```diff
@@ -43,9 +45,24 @@ export default class DatePicker {
    * @param {{yyyy?: number, month?: number, dayOfMonth?: number}} opts
    * @returns {Promise<string>}
    */
-  public async pickDate(opts: { yyyy?: number, month?: number, dayOfMonth?: number } = {}): Promise<string> {
+  public async pickDate(opts: {
+    yyyy?: number,
+    month?: number,
+    dayOfMonth?: number,
+    isToday?: boolean,
```
Comment (aweng98) [inline playwright-e2e/dsm/component/date-picker.ts:52]: added new optional param `isToday` to allow click of Today button.

File: playwright-e2e/tests/dsm/medical-record-request.spec.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+      await test.step('Find an enrolled participant', async () => {
+        const registrationDateColumn = 'Registration Date';
+        // These are columns displayed in Medical Records table
+        const typeColumn = 'Type';
+        const institutionNameColumn = 'Institution Name';
+        const followUpColumn = 'Follow-Up required';
+        const mrReviewColumn = 'MR Review';
+        const mrProblemColumn = 'MR Review';
+        const initialMRRequestColumn = 'Initial MR Request';
+        const initialMRReceivedColumn = 'Initial MR Received';
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/medical-record-request.spec.ts]: columns used to find a suitable participant whom is Enrolled, has medical records table and does not have an Initial MR Received date.

File: playwright-e2e/dsm/component/tabs/survey-data-tab.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Locator, Page, expect } from '@playwright/test';
+import { SurveyDataPanelEnum } from './enums/survey-data-enum';
+
+export default class SurveyDataTab {
+  constructor(private readonly page: Page) {
+  }
+
+  public async getActivityData(activity: SurveyDataPanelEnum, label: string): Promise<string[]> {
+    await this.activityDataPanel(activity).waitFor({state: 'visible' });
+    const region = this.activityDataPanel(activity).locator('xpath=//*[@role="region"]');
```
Comment (Rinchan20) [inline playwright-e2e/dsm/component/tabs/survey-data-tab.ts:10]: nice way to make locators neater (using a private method) + add locator method to it

File: playwright-e2e/tests/dsm/medical-record-request.spec.ts [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    'Asante Three Rivers Medical Center',
+    'Bothwell Regional Health Center',
+    'Adcare Hospital of Worcester Inc',
+    'Brigham and Women\'s',
+    'Asante Three Rivers Medical Center',
+    'Boston Children\'s Hospital'
+  ];
+
+  // Two CMI Clinical and two CMI Research studies
+  const studies: StudyEnum[] = [StudyEnum.OSTEO2, StudyEnum.LMS, StudyEnum.PANCAN, StudyEnum.BRAIN];
```
Comment (Rinchan20) [inline playwright-e2e/tests/dsm/medical-record-request.spec.ts:33]: nice!

**Comment by aweng98** [comment]
Updated Search criteria.

**Review by Rinchan20 — APPROVED** [review]
runs well in dev for me

**Pull request merged**